observers/slack_resource.cpp

```diff
@@ -0,0 +1,109 @@
+#include <memory>
+#include <vector>
+
+#include "glog/logging.h"
+
+#include "mesos/mesos.hpp"
+
+#include "serenity/executor_set.hpp"
+#include "serenity/metrics_helpers.hpp"
+
+#include "observers/slack_resource.hpp"
+
+namespace mesos {
+namespace serenity {
+
+Try<Nothing> SlackResourceObserver::consume(const ResourceUsage& usage) {
+  std::unique_ptr<ExecutorSet> newSamples(new ExecutorSet());
+  std::vector<Resource> slackResources;
+
+  for (const auto& executor : usage.executors()) {
+    if (executor.has_statistics() && executor.has_executor_info()) {
+      newSamples->insert(executor);
+
+      auto previousSample = this->previousSamples->find(executor);
+      if (previousSample != this->previousSamples->end()) {
+        Result<Resource> slackResource = CalculateSlack(
+            (*previousSample), executor);
+        if (slackResource.isSome()) {
+          slackResources.push_back(slackResource.get());
```

Comment:
I would like this class to observe all classes of slack resources. The Resource object should of different types stored in 'slackResources' vector should be later processed in 'CombineSlack' by grouping and flattening.